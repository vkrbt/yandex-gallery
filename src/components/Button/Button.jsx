import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export const Button = props => <button {...props} className={classnames('button', props.className)} />;

Button.propTypes = {
  className: PropTypes.string,
}
