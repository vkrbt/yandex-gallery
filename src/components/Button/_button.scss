.button {
  border: none;
  outline: none;
  background: transparent;
}
