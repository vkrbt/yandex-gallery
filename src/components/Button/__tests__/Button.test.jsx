import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '../Button';

describe('Button', () => {
  it('should being rendered normally', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should has className if it provided', () => {
    const className = 'wonderfulClassName';
    const wrapper = shallow(<Button className={className} />);
    expect(wrapper.hasClass(className)).toBeTruthy();
  });
});
