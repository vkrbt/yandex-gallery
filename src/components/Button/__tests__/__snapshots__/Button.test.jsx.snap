// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should being rendered normally 1`] = `
<button
  className="button"
/>
`;
